### PR TITLE
Bug 1262394 - [TV][Home] Sort cards by group-alphabet order in card-picker

### PR DESCRIPTION
**Pull request by luke-chang**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3efe150665c61f3ad70ee41c14b28de399d952bb

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0fbb29cea8d5a5936f6cfeb36df31f05a2bf470b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c7e5650b2afc25e21b050a0e4a243abe4b77c93

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=416f37ed8eba03aea447f209d7b88ca0044e5f4f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5eac7ebec6d66235a2acc197173a58074cc758a5

File: tv_apps/smart-home/js/card_picker.js [hunk 1]
```diff
@@ -114,8 +120,21 @@
...
         var appButton = CardUtil.createCardButton(card, true);
-        that.gridView.appendChild(appButton);
-        that.appButtons.push(appButton);
+        var key = String(defaultOrder[card.group] || 5);
+        if (!(card instanceof Deck)) {
+          var lang = document.documentElement.lang;
+          var name = that._cardManager.resolveCardName(card, lang);
+          if (name.raw) {
+            key += name.raw;
+          } else if (name.args && name.args.number !== undefined) {
```
Comment (rexboy7) [inline tv_apps/smart-home/js/card_picker.js]: Please describe it explicitly the purpose of this if-condition that it's for TV cards.

File: tv_apps/smart-home/js/card_picker.js [hunk 2]
```diff
@@ -114,8 +120,21 @@
...
+            key += '0'.repeat(10 - name.args.number.length) + name.args.number;
+          } else if (name.id) {
+            key += name.id;
```
Comment (rexboy7) [inline tv_apps/smart-home/js/card_picker.js]: ...and also for this (as we use id as a simplified key for sorting)

File: tv_apps/smart-home/js/card_picker.js [hunk 3]
```diff
@@ -114,8 +120,21 @@
...
+          } else {
+            key += '_';
```
Comment (rexboy7) [inline tv_apps/smart-home/js/card_picker.js]: And I don't really sure what underline here means so please describe it.

Comment (rexboy7) [inline tv_apps/smart-home/js/card_picker.js]: Due to spec Jack updated to me a few days ago, we should sort only those cards in cardlist. (e.g. cards in folderList don't need to be sorted)

And by the current design, if we don't put `folder` cards together in front of the list, things would get wrong. Maybe this implication is not a good design so you can consider changing `every` to `forEach` here https://github.com/mozilla-b2g/gaia/pull/34241/files#diff-3d7aba178b4738a2cba1c8a8ab1f40e1R186

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c846949ca64f0a95b03eb23b1e239330c2b1287

**Pull request merged**